eng/tools/generator/typespec/tspconfig.go

```diff
@@ -0,0 +1,146 @@
+	Imports              string         `yaml:"imports,omitempty"`
+	Emit                 []string       `yaml:"emit,omitempty"`
+	Options              map[string]any `yaml:"options,omitempty"`
+	Linter               LinterConfig   `yaml:"linter,omitempty"`
+}
+
+// <library name>:<rule/ruleset name>
+type LinterConfig struct {
+	Extends []RuleRef          `yaml:"extends,omitempty"`
+	Enable  map[RuleRef]bool   `yaml:"enable,omitempty"`
+	Disable map[RuleRef]string `yaml:"disable,omitempty"`
+}
+
+type RuleRef string
+
+func (r RuleRef) Validate() bool {
+	return regexp.MustCompile(`.*/.*`).MatchString(string(r))
+}
+
+func ParseTypeSpecConfig(tspconfigPath string) (*TypeSpecConfig, error) {
+	tspConfig := TypeSpecConfig{}
+	tspConfig.Path = tspconfigPath
+
+	var err error
+	var data []byte
+	if strings.HasPrefix(tspconfigPath, "http") {
+		// http path
+		resp, err := http.Get(tspconfigPath)
+		if err != nil {
+			return nil, err
+		}
+		defer resp.Body.Close()
+		data, err = io.ReadAll(resp.Body)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		// local path
+		data, err = os.ReadFile(tspconfigPath)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	err = yaml.Unmarshal(data, &(tspConfig.TypeSpecProjectSchema))
+	if err != nil {
+		return nil, err
+	}
+
+	return &tspConfig, err
+}
+
+func (tc *TypeSpecConfig) EditOptions(emit string, option map[string]any, append bool) {
+	if tc.Options == nil {
+		tc.Options = make(map[string]any)
+	}
+
+	if _, ok := tc.Options[emit]; ok {
+		if append {
+			op1 := tc.Options[emit].(map[string]any)
+			for k, v := range option {
+				op1[k] = v
+			}
+			tc.Options[emit] = op1
+		} else {
+			tc.Options[emit] = option
+		}
+	} else {
+		tc.Options[emit] = option
+	}
+}
+
+func (tc *TypeSpecConfig) Save() error {
+	data, err := yaml.MarshalWithOptions(tc.TypeSpecProjectSchema, yaml.IndentSequence(true))
+	if err != nil {
+		return err
+	}
+
+	return os.WriteFile(tc.Path, data, 0666)
+}
+
+func (tc *TypeSpecConfig) EmitOption(emit string) (any, error) {
+	if tc.Options == nil {
+		return nil, fmt.Errorf("no options found in %s", tc.Path)
+	}
+
+	if v, ok := tc.Options[emit]; ok {
+		return v, nil
+	}
+
+	return nil, fmt.Errorf("emit %s not found in %s", emit, tc.Path)
+}
+
+func (tc TypeSpecConfig) ExistEmitOption(emit string) bool {
+	_, err := tc.EmitOption(emit)
+	return err == nil
+}
+
+func (tc TypeSpecConfig) GetModuleName() ([2]string, error) {
+	option, err := tc.EmitOption(string(TypeSpec_GO))
+	if err != nil {
+		return [2]string{}, err
+	}
+
+	module := (option.(map[string]any))["module"].(string)
+	s := strings.Split(module, "/")
+	l := len(s)
+	if l != 7 {
+		return [2]string{}, fmt.Errorf("module is invalid and must be in the format of `github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/{rpName}/{packageName}`")
+	}
+	if !strings.Contains(s[l-1], "arm") {
+		return [2]string{}, fmt.Errorf("packageName is invalid and must start with `arm`")
+	}
+
+	return [2]string{s[l-2], s[l-1]}, nil
```

Comment:
Can you add a comment and example of what format will be returned?

Reply:
exmpales return: [scvmm, armscvmm]

Reply:
Can you add this example of return value as comment above the code to improve the readability?